code/go/0chain.net/chaincore/chain/worker.go

```diff
@@ -290,45 +286,61 @@ func (c *Chain) finalizeBlockProcess(ctx context.Context, fb *block.Block, bsh B
 		Logger.Info("finalize block - already finalized",
 			zap.Int64("round", fb.Round),
 			zap.String("block", fb.Hash))
-		return
+		return nil
 	}
 
 	Logger.Debug("start to finalize block",
 		zap.Int64("round", fb.Round),
 		zap.String("block", fb.Hash),
 		zap.String("prev block", fb.PrevHash))
 
+	isSharder := node.Self.IsSharder()
```

Comment:
this code should be moved to miner/sharder packages perhaps, and chaincore/chain should not know anything about difference between miners and sharders, but it's easy said than done. lgtm for now, maybe can refactor it after mainnet